{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting GEOparse\n",
      "  Downloading GEOparse-2.0.3.tar.gz (278 kB)\n",
      "\u001b[K     |████████████████████████████████| 278 kB 1.8 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: numpy>=1.7 in /home/adit/anaconda3/lib/python3.7/site-packages (from GEOparse) (1.19.2)\n",
      "Requirement already satisfied: pandas>=0.17 in /home/adit/.local/lib/python3.7/site-packages (from GEOparse) (1.0.3)\n",
      "Requirement already satisfied: requests>=2.21.0 in /home/adit/.local/lib/python3.7/site-packages (from GEOparse) (2.23.0)\n",
      "Requirement already satisfied: tqdm>=4.31.1 in /home/adit/anaconda3/lib/python3.7/site-packages (from GEOparse) (4.54.1)\n",
      "Requirement already satisfied: python-dateutil>=2.6.1 in /home/adit/anaconda3/lib/python3.7/site-packages (from pandas>=0.17->GEOparse) (2.8.1)\n",
      "Requirement already satisfied: pytz>=2017.2 in /home/adit/anaconda3/lib/python3.7/site-packages (from pandas>=0.17->GEOparse) (2020.4)\n",
      "Requirement already satisfied: chardet<4,>=3.0.2 in /home/adit/anaconda3/lib/python3.7/site-packages (from requests>=2.21.0->GEOparse) (3.0.4)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/adit/.local/lib/python3.7/site-packages (from requests>=2.21.0->GEOparse) (2019.11.28)\n",
      "Requirement already satisfied: idna<3,>=2.5 in /home/adit/.local/lib/python3.7/site-packages (from requests>=2.21.0->GEOparse) (2.9)\n",
      "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /home/adit/.local/lib/python3.7/site-packages (from requests>=2.21.0->GEOparse) (1.25.8)\n",
      "Requirement already satisfied: six>=1.5 in /home/adit/anaconda3/lib/python3.7/site-packages (from python-dateutil>=2.6.1->pandas>=0.17->GEOparse) (1.15.0)\n",
      "Building wheels for collected packages: GEOparse\n",
      "  Building wheel for GEOparse (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for GEOparse: filename=GEOparse-2.0.3-py3-none-any.whl size=29032 sha256=5f3ede31684474614f7db3d4cc768814081f9aa461a47fec3a793faa683685b5\n",
      "  Stored in directory: /home/adit/.cache/pip/wheels/4d/15/e8/fbf3b47444215d9728c20d7b35436b50086aa67c2ad6dcedad\n",
      "Successfully built GEOparse\n",
      "Installing collected packages: GEOparse\n",
      "Successfully installed GEOparse-2.0.3\n",
      "\u001b[33mWARNING: You are using pip version 20.0.2; however, version 21.2.4 is available.\n",
      "You should consider upgrading via the '/home/adit/anaconda3/bin/python -m pip install --upgrade pip' command.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!pip install GEOparse\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "import GEOparse\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "gse = GEOparse.get_GEO(filepath=\"./GSE7390_family.soft.gz\",silent=True,)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GSM example:\n",
      "Name:  GSM177885\n",
      "Metadata:\n",
      " - title : VDXGUYU_4002\n",
      " - geo_accession : GSM177885\n",
      " - status : Public on Jun 11 2007\n",
      " - submission_date : Mar 28 2007\n",
      " - last_update_date : May 31 2013\n",
      " - type : RNA\n",
      " - channel_count : 1\n",
      " - source_name_ch1 : VDXGUYU_4002\n",
      " - organism_ch1 : Homo sapiens\n",
      " - taxid_ch1 : 9606\n",
      " - characteristics_ch1 : samplename: VDXGUYU_4002, id: 4002, filename: 10-4002_GUYA.CEL.gz, hospital: GUY, age: 57, size: 3, Surgery_type: 0, Histtype: 1, Angioinv: 1, Lymp_infil: 2, node: 0, grade: 3, er: 0, t.rfs: 723, e.rfs: 1, t.os: 937, e.os: 1, t.dmfs: 723, e.dmfs: 1, t.tdm: 723, e.tdm: 1, risksg: 2, NPI: 4.6, risknpi: 2, AOL_os_10y: 62.7, risk_AOL: 1, veridex_risk: Poor\n",
      " - molecule_ch1 : total RNA\n",
      " - extract_protocol_ch1 : Isolation of RNA was performed using the Trizol method (Invitrogen) according to the manufacturers instructions and was purified using RNeasy mini-columns (Qiagen, Valencia, CA). The quality of the RNA obtained from each tumour sample was assessed based on the RNA profile generated by the bioanalyzer (Agilent Inc)olation of RNA was performed using the Trizol method (Invitrogen) according to the manufacturers instructions and was purified using RNeasy mini-columns (Qiagen, Valencia, CA). The quality of the RNA obtained from each tumour sample was assessed based on the RNA profile generated by the bioanalyzer (Agilent Inc).\n",
      " - label_ch1 : biotin\n",
      " - label_protocol_ch1 : Approximately 1 microgram of total RNA was processed to produce biotinylated cRNA targets.\n",
      " - hyb_protocol : standard Affymetrix procedures.\n",
      " - scan_protocol : standard Affymetrix procedures.\n",
      " - description : n/a\n",
      " - data_processing : MAS5 Bioconductor (target value of 600).\n",
      " - platform_id : GPL96\n",
      " - contact_name : Benjamin,,Haibe-Kains\n",
      " - contact_email : benjamin.haibe.kains@utoronto.ca\n",
      " - contact_phone : +14165818626\n",
      " - contact_laboratory : Bioinformatics and Computational Genomics\n",
      " - contact_department : Princess Margaret Research\n",
      " - contact_institute : Princess Margaret Cancer Centre\n",
      " - contact_address : 610 University Avenue\n",
      " - contact_city : Toronto\n",
      " - contact_state : Ontario\n",
      " - contact_zip/postal_code : M5G 2M9\n",
      " - contact_country : Canada\n",
      " - supplementary_file : ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM177nnn/GSM177885/suppl/GSM177885.cel.gz\n",
      " - relation : Reanalyzed by: GSE31519, Reanalyzed by: GSE47561\n",
      " - series_id : GSE7390\n",
      " - data_row_count : 22283\n",
      "Table data:\n",
      "                ID_REF     VALUE\n",
      "22278   AFFX-ThrX-5_at  5.680176\n",
      "22279   AFFX-ThrX-M_at  1.753631\n",
      "22280  AFFX-TrpnX-3_at  2.223181\n",
      "22281  AFFX-TrpnX-5_at  4.013873\n",
      "22282  AFFX-TrpnX-M_at  1.603252\n",
      "\n",
      "GPL example:\n",
      "Name:  GPL96\n",
      "Metadata:\n",
      " - title : [HG-U133A] Affymetrix Human Genome U133A Array\n",
      " - geo_accession : GPL96\n",
      " - status : Public on Mar 11 2002\n",
      " - submission_date : Feb 19 2002\n",
      " - last_update_date : Mar 09 2021\n",
      " - technology : in situ oligonucleotide\n",
      " - distribution : commercial\n",
      " - organism : Homo sapiens\n",
      " - taxid : 9606\n",
      " - manufacturer : Affymetrix\n",
      " - manufacture_protocol : see manufacturer's web site, , The U133 set includes 2 arrays with a total of 44928 entries and was indexed 29-Jan-2002., The set includes over 1,000,000 unique oligonucleotide features covering more than 39,000 transcript variants, which in turn represent greater than 33,000 of the best characterized human genes., Sequences were selected from GenBank, dbEST, and RefSeq. Sequence clusters were created from Build 133 of UniGene (April 20, 2001) and refined by analysis and comparison with a number of other publicly available databases including the Washington University EST trace repository and the University of California, Santa Cruz golden-path human genome database (April 2001 release). In addition, ESTs were analyzed for untrimmed low-quality sequence information, correct orientation, false priming, false clustering, alternative splicing and alternative polyadenylation., \n",
      " - description : Affymetrix submissions are typically submitted to GEO using the GEOarchive method described at http://www.ncbi.nlm.nih.gov/projects/geo/info/geo_affy.html, , June 03, 2009: annotation table updated with netaffx build 28, June 08, 2012: annotation table updated with netaffx build 32, June 24, 2016: annotation table updated with netaffx build 35\n",
      " - web_link : http://www.affymetrix.com/support/technical/byproduct.affx?product=hgu133, http://www.affymetrix.com/analysis/index.affx\n",
      " - contact_name : ,,Affymetrix, Inc.\n",
      " - contact_email : geo@ncbi.nlm.nih.gov, support@affymetrix.com\n",
      " - contact_phone : 888-362-2447\n",
      " - contact_institute : Affymetrix, Inc.\n",
      " - contact_address : \n",
      " - contact_city : Santa Clara\n",
      " - contact_state : CA\n",
      " - contact_zip/postal_code : 95051\n",
      " - contact_country : USA\n",
      " - contact_web_link : http://www.affymetrix.com/index.affx\n",
      " - relation : Alternative to: GPL4557 (Alternative CDF), Alternative to: GPL7566 (Alternative CDF), Alternative to: GPL9021 (Alternative CDF), Alternative to: GPL9741 (Alternative CDF), Alternative to: GPL10553 (Alternative CDF: ncilpg_U133Agb), Alternative to: GPL14663 (Alternative CDF), Alternative to: GPL16212 (Alternative CDF), Alternative to: GPL17027 (Alternative CDF), Alternative to: GPL17662 (alternative CDF), Alternative to: GPL19184 (CDF: v18 ENTREZG), Alternative to: GPL29829 (gene name)\n",
      " - data_row_count : 22283\n",
      "Table data:\n",
      "          ID  GB_ACC SPOT_ID Species Scientific Name Annotation Date  \\\n",
      "0  1007_s_at  U48705     NaN            Homo sapiens     Oct 6, 2014   \n",
      "1    1053_at  M87338     NaN            Homo sapiens     Oct 6, 2014   \n",
      "2     117_at  X51757     NaN            Homo sapiens     Oct 6, 2014   \n",
      "3     121_at  X69699     NaN            Homo sapiens     Oct 6, 2014   \n",
      "4  1255_g_at  L36861     NaN            Homo sapiens     Oct 6, 2014   \n",
      "\n",
      "       Sequence Type                  Sequence Source  \\\n",
      "0  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "1  Exemplar sequence                          GenBank   \n",
      "2  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "3  Exemplar sequence                          GenBank   \n",
      "4  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "\n",
      "                                  Target Description Representative Public ID  \\\n",
      "0  U48705 /FEATURE=mRNA /DEFINITION=HSU48705 Huma...                   U48705   \n",
      "1  M87338 /FEATURE= /DEFINITION=HUMA1SBU Human re...                   M87338   \n",
      "2  X51757 /FEATURE=cds /DEFINITION=HSP70B Human h...                   X51757   \n",
      "3  X69699 /FEATURE= /DEFINITION=HSPAX8A H.sapiens...                   X69699   \n",
      "4  L36861 /FEATURE=expanded_cds /DEFINITION=HUMGC...                   L36861   \n",
      "\n",
      "                                          Gene Title       Gene Symbol  \\\n",
      "0  discoidin domain receptor tyrosine kinase 1 //...  DDR1 /// MIR4640   \n",
      "1        replication factor C (activator 1) 2, 40kDa              RFC2   \n",
      "2               heat shock 70kDa protein 6 (HSP70B')             HSPA6   \n",
      "3                                       paired box 8              PAX8   \n",
      "4            guanylate cyclase activator 1A (retina)            GUCA1A   \n",
      "\n",
      "      ENTREZ_GENE_ID                               RefSeq Transcript ID  \\\n",
      "0  780 /// 100616237  NM_001202521 /// NM_001202522 /// NM_001202523...   \n",
      "1               5982  NM_001278791 /// NM_001278792 /// NM_001278793...   \n",
      "2               3310                                          NM_002155   \n",
      "3               7849  NM_003466 /// NM_013951 /// NM_013952 /// NM_0...   \n",
      "4               2978                         NM_000409 /// XM_006715073   \n",
      "\n",
      "                    Gene Ontology Biological Process  \\\n",
      "0  0001558 // regulation of cell growth // inferr...   \n",
      "1  0000278 // mitotic cell cycle // traceable aut...   \n",
      "2  0000902 // cell morphogenesis // inferred from...   \n",
      "3  0001655 // urogenital system development // in...   \n",
      "4  0007165 // signal transduction // non-traceabl...   \n",
      "\n",
      "                    Gene Ontology Cellular Component  \\\n",
      "0  0005576 // extracellular region // inferred fr...   \n",
      "1  0005634 // nucleus // inferred from electronic...   \n",
      "2  0005737 // cytoplasm // inferred from direct a...   \n",
      "3  0005634 // nucleus // inferred from direct ass...   \n",
      "4  0001750 // photoreceptor outer segment // infe...   \n",
      "\n",
      "                    Gene Ontology Molecular Function  \n",
      "0  0000166 // nucleotide binding // inferred from...  \n",
      "1  0000166 // nucleotide binding // inferred from...  \n",
      "2  0000166 // nucleotide binding // inferred from...  \n",
      "3  0000979 // RNA polymerase II core promoter seq...  \n",
      "4  0005509 // calcium ion binding // inferred fro...  \n"
     ]
    }
   ],
   "source": [
    "print()\n",
    "print(\"GSM example:\")\n",
    "for gsm_name, gsm in gse.gsms.items():\n",
    "    print(\"Name: \", gsm_name)\n",
    "    print(\"Metadata:\",)\n",
    "    for key, value in gsm.metadata.items():\n",
    "        print(\" - %s : %s\" % (key, \", \".join(value)))\n",
    "    print (\"Table data:\",)\n",
    "    print (gsm.table.tail())\n",
    "    break\n",
    "\n",
    "print()\n",
    "print(\"GPL example:\")\n",
    "for gpl_name, gpl in gse.gpls.items():\n",
    "    print(\"Name: \", gpl_name)\n",
    "    print(\"Metadata:\",)\n",
    "    for key, value in gpl.metadata.items():\n",
    "        print(\" - %s : %s\" % (key, \", \".join(value)))\n",
    "    print(\"Table data:\",)\n",
    "    print(gpl.table.head())\n",
    "    break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <td>Affymetrix Probe Set ID LINK_PRE:\"https://www....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GB_ACC</th>\n",
       "      <td>GenBank Accession Number LINK_PRE:\"http://www....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPOT_ID</th>\n",
       "      <td>identifies controls</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Species Scientific Name</th>\n",
       "      <td>The genus and species of the organism represen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Annotation Date</th>\n",
       "      <td>The date that the annotations for this probe a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sequence Type</th>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sequence Source</th>\n",
       "      <td>The database from which the sequence used to d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Target Description</th>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Representative Public ID</th>\n",
       "      <td>The accession number of a representative seque...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Title</th>\n",
       "      <td>Title of Gene represented by the probe set.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Symbol</th>\n",
       "      <td>A gene symbol, when one is available (from Uni...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENTREZ_GENE_ID</th>\n",
       "      <td>Entrez Gene Database UID LINK_PRE:\"http://www....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RefSeq Transcript ID</th>\n",
       "      <td>References to multiple sequences in RefSeq. Th...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Ontology Biological Process</th>\n",
       "      <td>Gene Ontology Consortium Biological Process de...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Ontology Cellular Component</th>\n",
       "      <td>Gene Ontology Consortium Cellular Component de...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Ontology Molecular Function</th>\n",
       "      <td>Gene Ontology Consortium Molecular Function de...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                        description\n",
       "ID                                Affymetrix Probe Set ID LINK_PRE:\"https://www....\n",
       "GB_ACC                            GenBank Accession Number LINK_PRE:\"http://www....\n",
       "SPOT_ID                                                         identifies controls\n",
       "Species Scientific Name           The genus and species of the organism represen...\n",
       "Annotation Date                   The date that the annotations for this probe a...\n",
       "Sequence Type                                                                      \n",
       "Sequence Source                   The database from which the sequence used to d...\n",
       "Target Description                                                                 \n",
       "Representative Public ID          The accession number of a representative seque...\n",
       "Gene Title                              Title of Gene represented by the probe set.\n",
       "Gene Symbol                       A gene symbol, when one is available (from Uni...\n",
       "ENTREZ_GENE_ID                    Entrez Gene Database UID LINK_PRE:\"http://www....\n",
       "RefSeq Transcript ID              References to multiple sequences in RefSeq. Th...\n",
       "Gene Ontology Biological Process  Gene Ontology Consortium Biological Process de...\n",
       "Gene Ontology Cellular Component  Gene Ontology Consortium Cellular Component de...\n",
       "Gene Ontology Molecular Function  Gene Ontology Consortium Molecular Function de..."
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse.gsms['GPL96'].columns\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['samplename: VDXGUYU_4002',\n",
       " 'id: 4002',\n",
       " 'filename: 10-4002_GUYA.CEL.gz',\n",
       " 'hospital: GUY',\n",
       " 'age: 57',\n",
       " 'size: 3',\n",
       " 'Surgery_type: 0',\n",
       " 'Histtype: 1',\n",
       " 'Angioinv: 1',\n",
       " 'Lymp_infil: 2',\n",
       " 'node: 0',\n",
       " 'grade: 3',\n",
       " 'er: 0',\n",
       " 't.rfs: 723',\n",
       " 'e.rfs: 1',\n",
       " 't.os: 937',\n",
       " 'e.os: 1',\n",
       " 't.dmfs: 723',\n",
       " 'e.dmfs: 1',\n",
       " 't.tdm: 723',\n",
       " 'e.tdm: 1',\n",
       " 'risksg: 2',\n",
       " 'NPI: 4.6',\n",
       " 'risknpi: 2',\n",
       " 'AOL_os_10y: 62.7',\n",
       " 'risk_AOL: 1',\n",
       " 'veridex_risk: Poor']"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse.gsms['GSM177885'].metadata['characteristics_ch1']#.GPL96"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID_REF</th>\n",
       "      <th>VALUE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1007_s_at</td>\n",
       "      <td>10.780787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1053_at</td>\n",
       "      <td>8.674201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>117_at</td>\n",
       "      <td>7.738589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>121_at</td>\n",
       "      <td>9.285511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1255_g_at</td>\n",
       "      <td>6.610484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22278</th>\n",
       "      <td>AFFX-ThrX-5_at</td>\n",
       "      <td>5.680176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22279</th>\n",
       "      <td>AFFX-ThrX-M_at</td>\n",
       "      <td>1.753631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22280</th>\n",
       "      <td>AFFX-TrpnX-3_at</td>\n",
       "      <td>2.223181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22281</th>\n",
       "      <td>AFFX-TrpnX-5_at</td>\n",
       "      <td>4.013873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22282</th>\n",
       "      <td>AFFX-TrpnX-M_at</td>\n",
       "      <td>1.603252</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22283 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ID_REF      VALUE\n",
       "0            1007_s_at  10.780787\n",
       "1              1053_at   8.674201\n",
       "2               117_at   7.738589\n",
       "3               121_at   9.285511\n",
       "4            1255_g_at   6.610484\n",
       "...                ...        ...\n",
       "22278   AFFX-ThrX-5_at   5.680176\n",
       "22279   AFFX-ThrX-M_at   1.753631\n",
       "22280  AFFX-TrpnX-3_at   2.223181\n",
       "22281  AFFX-TrpnX-5_at   4.013873\n",
       "22282  AFFX-TrpnX-M_at   1.603252\n",
       "\n",
       "[22283 rows x 2 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse.gsms['GSM177885'].table#.data#['characteristics_ch1']#.GPL96"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
